okta_request_condition panic: runtime error: invalid memory address or nil pointer dereference

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
        Before submitting a bug report, we ask that you first [search existing issues](https://github.com/okta/terraform-provider-okta/issues) and [pull requests](https://github.com/okta/terraform-provider-okta/pulls) to see if someone else may have experienced the same issue or may have already submitted a fix for it. This helps to keep all relevant information in one place, including any potential workarounds.

### A Note on Terraform Core Issues

We also ask that you consider whether your issue may be related to Terraform Core. If you are running into one of the following scenarios, we recommend [opening an issue](https://github.com/hashicorp/terraform/issues/new/choose) in the Terraform Core repository instead:

* [Configuration Language](https://developer.hashicorp.com/terraform/language) or resource ordering issues
* [State](https://developer.hashicorp.com/terraform/language/state) and [State Backend](https://developer.hashicorp.com/terraform/language/backend) issues
* [Provisioner](https://developer.hashicorp.com/terraform/language/resources/provisioners/syntax) issues
* [Registry](https://registry.terraform.io/) issues
* Issues that span resources across multiple providers


### Terraform Version & Okta Provider Version(s)

Terraform v1.13.3
on darwin_arm64
+ provider registry.terraform.io/providers/okta/okta/ v6.3.0


### Affected Resource(s)

okta_request_condition

### Can this be done in the Admin UI?

No

### Can this be done in the actual API call?

Yes

### Customer Information

Organization Name: 10x Genomics
Paid Customer: yes


### Terraform Configuration

```tf
resource "okta_request_condition" "dylan_test_sso_app" {
  resource_id          = local.dylan_test_app
  approval_sequence_id = local.approval_manager
  name                 = "App Request Test"

  access_scope_settings {
    type = "GROUPS"
    ids {
      id = local.dylan_test_group
    }
  }
  requester_settings {
    type = "EVERYONE"
  }
}

```


### Debug Output

```text
Stack trace from the terraform-provider-okta_v6.3.0 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x10222d230]

goroutine 86 [running]:
github.com/okta/terraform-provider-okta/okta/services/governance.createRequestCondition({{0x1, {0x0, 0x0}}, {0x2, {0x14000119500, 0x14}}, {0x2, {0x140001194d0, 0x18}}, {0x2, ...}, ...})
        github.com/okta/terraform-provider-okta/okta/services/governance/resource_request_condition.go:404 +0x1a0
github.com/okta/terraform-provider-okta/okta/services/governance.(*requestConditionResource).Create(0x14000526030, {0x102d10f18, 0x1400071f980}, {{{{0x102d17918, 0x14000657b90}, {0x102934bc0, 0x1400064d920}}, {0x102d1b2d8, 0x1400042fbd0}}, {{{0x102d17918, ...}, ...}, ...}, ...}, ...)
        github.com/okta/terraform-provider-okta/okta/services/governance/resource_request_condition.go:190 +0x130
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).CreateResource(0x1400016b688, {0x102d10f18, 0x1400071f980}, 0x140005d1390, 0x140005d1360)
        github.com/hashicorp/terraform-plugin-framework@v1.15.1/internal/fwserver/server_createresource.go:127 +0x648
github.com/hashicorp/terraform-plugin-framework/internal/fwserver.(*Server).ApplyResourceChange(0x1400016b688, {0x102d10f18, 0x1400071f980}, 0x1400005e540, 0x140005d14b8)
        github.com/hashicorp/terraform-plugin-framework@v1.15.1/internal/fwserver/server_applyresourcechange.go:63 +0x434
github.com/hashicorp/terraform-plugin-framework/internal/proto6server.(*Server).ApplyResourceChange(0x1400016b688, {0x102d10f18?, 0x1400071f7d0?}, 0x14000598af0)
        github.com/hashicorp/terraform-plugin-framework@v1.15.1/internal/proto6server/server_applyresourcechange.go:71 +0x434
github.com/hashicorp/terraform-plugin-mux/tf6to5server.v6tov5Server.ApplyResourceChange({{0x102d25188?, 0x1400016b688?}}, {0x102d10f18, 0x1400071f7d0}, 0x0?)
        github.com/hashicorp/terraform-plugin-mux@v0.20.0/tf6to5server/tf6to5server.go:50 +0x58
github.com/hashicorp/terraform-plugin-mux/tf5muxserver.(*muxServer).ApplyResourceChange(0x1400052a980, {0x102d10f18?, 0x1400071f500?}, 0x14000598aa0)
        github.com/hashicorp/terraform-plugin-mux@v0.20.0/tf5muxserver/mux_server_ApplyResourceChange.go:36 +0x184
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x140002b0320, {0x102d10f18?, 0x1400071e900?}, 0x1400060ea00)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/tf5server/server.go:941 +0x2a8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x102c62760, 0x140002b0320}, {0x102d10f18, 0x1400071e900}, 0x1400060e980, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.28.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:687 +0x1c0
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400029c000, {0x102d10f18, 0x1400071e870}, 0x140004ba120, 0x1400044f230, 0x103e7c208, 0x0)
        google.golang.org/grpc@v1.72.1/server.go:1405 +0xca8
google.golang.org/grpc.(*Server).handleStream(0x1400029c000, {0x102d11818, 0x14000344000}, 0x140004ba120)
        google.golang.org/grpc@v1.72.1/server.go:1815 +0x910
google.golang.org/grpc.(*Server).serveStreams.func2.1()
        google.golang.org/grpc@v1.72.1/server.go:1035 +0x84
created by google.golang.org/grpc.(*Server).serveStreams.func2 in goroutine 52
        google.golang.org/grpc@v1.72.1/server.go:1046 +0x13c

Error: The terraform-provider-okta_v6.3.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expected Behavior

The `terraform apply` should have run and created a request condition for an app with the provided access group.

### Actual Behavior

The Okta Terraform provider crashed

### Steps to reproduce

1. Create an `okta_request_condition` with `access_scope_settings` set to `GROUPS`
2. `terraform apply`

### Important Factoids

This bug only occurs when `access_scope_settings` is set to `GROUPS` or `ENTITLEMENT_BUNDLES`.

The bug is due to missing a few lines of code.
This is in the file [resource_request_condition.go](https://github.com/okta/terraform-provider-okta/blob/1efc38952e72cfae957bd00d9c03ff786996dffb/okta/services/governance/resource_request_condition.go)

I took a look at the code that works for [RESOURCE_DEFAULT](https://github.com/okta/terraform-provider-okta/blob/1efc38952e72cfae957bd00d9c03ff786996dffb/okta/services/governance/resource_request_condition.go#L427-L434) and it has extra code to create a reference to the struct so that it is not `nil`.
```go
		if accessScopeSettings.AccessScopeSettingsCreatableResourceDefaultAccessScopeSettings == nil {
			accessScopeSettings.AccessScopeSettingsCreatableResourceDefaultAccessScopeSettings =
				&governance.AccessScopeSettingsCreatableResourceDefaultAccessScopeSettings{}
		}
```

The code for `GROUPS` is missing that block [here](https://github.com/okta/terraform-provider-okta/blob/1efc38952e72cfae957bd00d9c03ff786996dffb/okta/services/governance/resource_request_condition.go#L404)
and `ENTITLEMENT_BUNDLES` [here](https://github.com/okta/terraform-provider-okta/blob/1efc38952e72cfae957bd00d9c03ff786996dffb/okta/services/governance/resource_request_condition.go#L416)

I believe the pointers are being created in [model_access_scope_settings_creatable_access_scope_settings.go](https://github.com/okta/okta-governance-sdk-golang/blob/9f0566c72ab95c453ba6d890c0b006971131631b/governance/model_access_scope_settings_creatable_access_scope_settings.go)

### References

[resource_request_condition.go](https://github.com/okta/terraform-provider-okta/blob/1efc38952e72cfae957bd00d9c03ff786996dffb/okta/services/governance/resource_request_condition.go)

[model_access_scope_settings_creatable_access_scope_settings.go](https://github.com/okta/okta-governance-sdk-golang/blob/9f0566c72ab95c453ba6d890c0b006971131631b/governance/model_access_scope_settings_creatable_access_scope_settings.go)

--- a/docs/resources/request_condition.md
+++ b/docs/resources/request_condition.md
@@ -56,13 +56,13 @@ resource "okta_request_condition" "example" {
 ### Nested Schema for `access_scope_settings`
 Required:
 - `type` (String) Enum: `RESOURCE_DEFAULT`, `GROUPS`, `ENTITLEMENT_BUNDLES`.
-- `id` (List of String) List of entitlement bundle ids/group ids.
+- `id` (Block) Block list of entitlement bundle ids/group ids.
 
 <a id="nestedblock--requester_settings"></a>
 ### Nested Schema for `requester_settings`
 Required:
 - `type` (String) Enum: `EVERYONE`, `TEAMS`, `GROUPS`.
-- `id` (List of String) List of group ids or team ids.
+- `id` (Block) Block list of group ids or team ids.
 
 <a id="nestedblock--access_duration_settings"></a>
 ### Nested Schema for `access_duration_settings`

--- a/examples/resources/okta_request_condition/basic_issue2510.tf
+++ b/examples/resources/okta_request_condition/basic_issue2510.tf
@@ -1,0 +1,26 @@
+resource "okta_group" "requester" {
+  name        = "requester_test"
+  description = "requester_test"
+}
+
+resource "okta_request_condition" "test" {
+  resource_id="0oaqgxmg2n2FjHLzw1d7"
+  approval_sequence_id="68d224058c0cff364ca377e8"
+  name="issue-2510"
+
+  access_scope_settings {
+    type = "GROUPS"
+
+    ids {
+      id = "00gouu5aq9Gq0JGLH1d7"
+    }
+  }
+
+  requester_settings {
+    type = "GROUPS"
+
+    ids {
+      id = okta_group.requester.id
+    }
+  }
+}

--- a/okta/services/governance/resource_request_condition.go
+++ b/okta/services/governance/resource_request_condition.go
@@ -187,22 +187,7 @@ func (r *requestConditionResource) Create(ctx context.Context, req resource.Crea
 		return
 	}
 
-	requestConditionReq, diags := createRequestCondition(data)
-	if diags.HasError() {
-		if diags.HasError() {
-			resp.Diagnostics.Append(diags...)
-			return
-		}
-	}
-	if requestConditionReq == nil {
-		resp.Diagnostics.AddError(
-			"Error creating Request conditions",
-			"Could not create Request conditions, unexpected error: requestConditionReq is nil",
-		)
-		return
-	}
-
-	requestConditionResp, _, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.CreateResourceRequestConditionV2(ctx, data.ResourceId.ValueString()).RequestConditionCreatable(*requestConditionReq).Execute()
+	requestConditionResp, _, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.CreateResourceRequestConditionV2(ctx, data.ResourceId.ValueString()).RequestConditionCreatable(createRequestCondition(data)).Execute()
 	if err != nil {
 		resp.Diagnostics.AddError(
 			"Error creating Request conditions",
@@ -211,7 +196,8 @@ func (r *requestConditionResource) Create(ctx context.Context, req resource.Crea
 		return
 	}
 
-	if applyRequestConditionToState(&data, requestConditionResp) {
+	resp.Diagnostics.Append(applyRequestConditionToState(ctx, &data, requestConditionResp)...)
+	if resp.Diagnostics.HasError() {
 		return
 	}
 
@@ -239,7 +225,10 @@ func (r *requestConditionResource) Read(ctx context.Context, req resource.ReadRe
 		return
 	}
 
-	applyRequestConditionToState(&data, readRequestConditionResp)
+	resp.Diagnostics.Append(applyRequestConditionToState(ctx, &data, readRequestConditionResp)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
@@ -258,15 +247,8 @@ func (r *requestConditionResource) Update(ctx context.Context, req resource.Upda
 		return
 	}
 
-	patch, diags := createRequestConditionPatch(data)
-	if diags.HasError() {
-		if diags.HasError() {
-			resp.Diagnostics.Append(diags...)
-			return
-		}
-	}
 	// Update API call logic
-	updatedRequestCondition, _, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.UpdateResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).RequestConditionPatchable(patch).Execute()
+	updatedRequestCondition, _, err := r.OktaGovernanceClient.OktaGovernanceSDKClient().RequestConditionsAPI.UpdateResourceRequestConditionV2(ctx, data.ResourceId.ValueString(), state.Id.ValueString()).RequestConditionPatchable(createRequestConditionPatch(data)).Execute()
 	if err != nil {
 		resp.Diagnostics.AddError(
 			"Error updating Request conditions",
@@ -275,8 +257,12 @@ func (r *requestConditionResource) Update(ctx context.Context, req resource.Upda
 		return
 	}
 
-	applyRequestConditionToState(&data, updatedRequestCondition)
-	// Save updated Data into Terraform state
+	resp.Diagnostics.Append(applyRequestConditionToState(ctx, &data, updatedRequestCondition)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save Data into Terraform state
 	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
 }
 
@@ -306,7 +292,8 @@ func (r *requestConditionResource) Delete(ctx context.Context, req resource.Dele
 	}
 }
 
-func applyRequestConditionToState(data *requestConditionResourceModel, requestConditionResp *governance.RequestConditionFull) bool {
+func applyRequestConditionToState(ctx context.Context, data *requestConditionResourceModel, requestConditionResp *governance.RequestConditionFull) diag.Diagnostics {
+	var diags diag.Diagnostics
 	data.Id = types.StringValue(requestConditionResp.GetId())
 	data.Name = types.StringValue(requestConditionResp.GetName())
 	if requestConditionResp.Description != nil {
@@ -322,7 +309,7 @@ func applyRequestConditionToState(data *requestConditionResourceModel, requestCo
 	data.RequesterSettings, _ = setRequesterSettings(requestConditionResp.GetRequesterSettings())
 	data.AccessScopeSettings, _ = setAccessScopeSettings(requestConditionResp.GetAccessScopeSettings())
 	data.AccessDurationSettings = setAccessDurationSettings(requestConditionResp.GetAccessDurationSettings())
-	return false
+	return diags
 }
 
 func setAccessDurationSettings(settings governance.AccessDurationSettingsFull) *AccessDurationSettings {
@@ -388,7 +375,7 @@ func setRequesterSettings(settings governance.RequesterSettingsFullRequesterSett
 	return &setting, nil
 }
 
-func createRequestCondition(data requestConditionResourceModel) (*governance.RequestConditionCreatable, diag.Diagnostics) {
+func createRequestCondition(data requestConditionResourceModel) governance.RequestConditionCreatable {
 	req := governance.RequestConditionCreatable{}
 	req.Name = data.Name.ValueString()
 	req.ApprovalSequenceId = data.ApprovalSequenceId.ValueString()
@@ -401,6 +388,10 @@ func createRequestCondition(data requestConditionResourceModel) (*governance.Req
 
 	accessScopeSettings := governance.AccessScopeSettingsCreatableAccessScopeSettings{}
 	if data.AccessScopeSettings.Type.ValueString() == "GROUPS" {
+		if accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings == nil {
+			accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings =
+				&governance.AccessScopeSettingsCreatableGroupAccessScopeSettings{}
+		}
 		accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings.Type = "GROUPS"
 		var groupsIds []governance.GroupsArrayCreatableInner
 		elems := data.AccessScopeSettings.Ids
@@ -413,6 +404,10 @@ func createRequestCondition(data requestConditionResourceModel) (*governance.Req
 		accessScopeSettings.AccessScopeSettingsCreatableGroupAccessScopeSettings.Groups = groupsIds
 		req.AccessScopeSettings = accessScopeSettings
 	} else if data.AccessScopeSettings.Type.ValueString() == "ENTITLEMENT_BUNDLES" {
+		if accessScopeSettings.AccessScopeSettingsCreatableEntitlementBundleAccessScopeSettings == nil {
+			accessScopeSettings.AccessScopeSettingsCreatableEntitlementBundleAccessScopeSettings =
+				&governance.AccessScopeSettingsCreatableEntitlementBundleAccessScopeSettings{}
+		}
 		accessScopeSettings.AccessScopeSettingsCreatableEntitlementBundleAccessScopeSettings.Type = "ENTITLEMENT_BUNDLES"
 		var entitlementBundles []governance.EntitlementBundlesArrayCreatableInner
 		elems := data.AccessScopeSettings.Ids
@@ -474,10 +469,10 @@ func createRequestCondition(data requestConditionResourceModel) (*governance.Req
 		}
 	}
 
-	return &req, nil
+	return req
 }
 
-func createRequestConditionPatch(data requestConditionResourceModel) (governance.RequestConditionPatchable, diag.Diagnostics) {
+func createRequestConditionPatch(data requestConditionResourceModel) governance.RequestConditionPatchable {
 	var patch governance.RequestConditionPatchable
 	patch.Name = data.Name.ValueStringPointer()
 	patch.ApprovalSequenceId = data.ApprovalSequenceId.ValueStringPointer()
@@ -567,5 +562,5 @@ func createRequestConditionPatch(data requestConditionResourceModel) (governance
 			patch.AccessDurationSettings.Set(&accessDurationSettings)
 		}
 	}
-	return patch, nil
+	return patch
 }

--- a/okta/services/governance/resource_request_condition_test.go
+++ b/okta/services/governance/resource_request_condition_test.go
@@ -36,5 +36,26 @@ func TestAccRequestConditionResource_basic(t *testing.T) {
 			},
 		},
 	})
+}
 
+func TestAccRequestConditionResource_Issue2510(t *testing.T) {
+	mgr := newFixtureManager("resources", resources.OktaGovernanceRequestCondition, t.Name())
+	config := mgr.GetFixtures("basic_issue2510.tf", t)
+	resourceName := fmt.Sprintf("%s.test", resources.OktaGovernanceRequestCondition)
+
+	acctest.OktaResourceTest(t, resource.TestCase{
+		PreCheck:                 acctest.AccPreCheck(t),
+		ErrorCheck:               testAccErrorChecks(t),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactoriesForTestAcc(t),
+		CheckDestroy:             nil,
+		Steps: []resource.TestStep{
+			{
+				Config: config,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, "name", "issue-2510"),
+					resource.TestCheckResourceAttr(resourceName, "requester_settings.type", "GROUPS"),
+				),
+			},
+		},
+	})
 }

--- a/test/fixtures/vcr/governance/TestAccRequestConditionResource_Issue2510/classic-00.yaml
+++ b/test/fixtures/vcr/governance/TestAccRequestConditionResource_Issue2510/classic-00.yaml
@@ -1,0 +1,106 @@
+---
+version: 2
+interactions:
+    - id: 0
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
+        content_length: 243
+        host: classic-00.dne-okta.com
+        body: |
+            {"accessScopeSettings":{"groups":[{"id":"00gouu5aq9Gq0JGLH1d7"}],"type":"GROUPS"},"approvalSequenceId":"68d224058c0cff364ca377e8","name":"issue-2510","priority":0,"requesterSettings":{"groups":[{"id":"00gqwtmv55uVcEBSi1d7"}],"type":"GROUPS"}}
+        headers:
+            Accept:
+                - application/json
+            Authorization:
+                - SSWS REDACTED
+            Content-Type:
+                - application/json
+        url: https://classic-00.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions
+        method: POST
+      response:
+        proto: HTTP/2.0
+        proto_major: 2
+        proto_minor: 0
+        content_length: 620
+        body: '{"requesterSettings":{"type":"GROUPS","groups":[{"id":"00gqwtmv55uVcEBSi1d7"}]},"accessScopeSettings":{"type":"GROUPS","groups":[{"id":"00gouu5aq9Gq0JGLH1d7"}]},"approvalSequenceId":"68d224058c0cff364ca377e8","name":"issue-2510","id":"rco12lq2otuPystzX1d7","createdBy":"00unkw1sfbTw08c0g1d7","created":"2025-10-12T21:11:00Z","lastUpdated":"2025-10-12T21:11:00Z","lastUpdatedBy":"00unkw1sfbTw08c0g1d7","_links":{"self":{"href":"https://classic-00-admin.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2otuPystzX1d7","hints":{}}},"status":"INACTIVE","priority":0}'
+        headers:
+            Accept-Ch:
+                - Sec-CH-UA-Platform-Version
+            Content-Length:
+                - "620"
+            Content-Type:
+                - application/json
+            Date:
+                - Sun, 12 Oct 2025 21:11:00 GMT
+            Location:
+                - https://dcp-testing-oig-2025-06-26-admin.oktapreview.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2otuPystzX1d7
+            Referrer-Policy:
+                - strict-origin-when-cross-origin
+        status: 201 Created
+        code: 201
+        duration: 2.498982125s
+    - id: 1
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
+        content_length: 0
+        host: classic-00.dne-okta.com
+        headers:
+            Accept:
+                - application/json
+            Authorization:
+                - SSWS REDACTED
+        url: https://classic-00.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2otuPystzX1d7
+        method: GET
+      response:
+        proto: HTTP/2.0
+        proto_major: 2
+        proto_minor: 0
+        content_length: -1
+        uncompressed: true
+        body: '{"requesterSettings":{"type":"GROUPS","groups":[{"id":"00gqwtmv55uVcEBSi1d7"}]},"accessScopeSettings":{"type":"GROUPS","groups":[{"id":"00gouu5aq9Gq0JGLH1d7"}]},"approvalSequenceId":"68d224058c0cff364ca377e8","name":"issue-2510","id":"rco12lq2otuPystzX1d7","createdBy":"00unkw1sfbTw08c0g1d7","created":"2025-10-12T21:11:00Z","lastUpdated":"2025-10-12T21:11:00Z","lastUpdatedBy":"00unkw1sfbTw08c0g1d7","_links":{"self":{"href":"https://classic-00-admin.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2otuPystzX1d7","hints":{}}},"status":"INACTIVE","priority":0}'
+        headers:
+            Accept-Ch:
+                - Sec-CH-UA-Platform-Version
+            Content-Type:
+                - application/json
+            Date:
+                - Sun, 12 Oct 2025 21:11:02 GMT
+            Referrer-Policy:
+                - strict-origin-when-cross-origin
+        status: 200 OK
+        code: 200
+        duration: 1.283624125s
+    - id: 2
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
+        content_length: 0
+        host: classic-00.dne-okta.com
+        headers:
+            Accept:
+                - application/json
+            Authorization:
+                - SSWS REDACTED
+        url: https://classic-00.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2otuPystzX1d7
+        method: DELETE
+      response:
+        proto: HTTP/2.0
+        proto_major: 2
+        proto_minor: 0
+        content_length: 0
+        body: ""
+        headers:
+            Accept-Ch:
+                - Sec-CH-UA-Platform-Version
+            Date:
+                - Sun, 12 Oct 2025 21:11:04 GMT
+            Referrer-Policy:
+                - strict-origin-when-cross-origin
+        status: 204 No Content
+        code: 204
+        duration: 1.3408165s

--- a/test/fixtures/vcr/governance/TestAccRequestConditionResource_Issue2510/oie-00.yaml
+++ b/test/fixtures/vcr/governance/TestAccRequestConditionResource_Issue2510/oie-00.yaml
@@ -1,0 +1,106 @@
+---
+version: 2
+interactions:
+    - id: 0
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
+        content_length: 243
+        host: oie-00.dne-okta.com
+        body: |
+            {"accessScopeSettings":{"groups":[{"id":"00gouu5aq9Gq0JGLH1d7"}],"type":"GROUPS"},"approvalSequenceId":"68d224058c0cff364ca377e8","name":"issue-2510","priority":0,"requesterSettings":{"groups":[{"id":"00gqwtobh7LL5TuLp1d7"}],"type":"GROUPS"}}
+        headers:
+            Accept:
+                - application/json
+            Authorization:
+                - SSWS REDACTED
+            Content-Type:
+                - application/json
+        url: https://oie-00.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions
+        method: POST
+      response:
+        proto: HTTP/2.0
+        proto_major: 2
+        proto_minor: 0
+        content_length: 620
+        body: '{"requesterSettings":{"type":"GROUPS","groups":[{"id":"00gqwtobh7LL5TuLp1d7"}]},"accessScopeSettings":{"type":"GROUPS","groups":[{"id":"00gouu5aq9Gq0JGLH1d7"}]},"approvalSequenceId":"68d224058c0cff364ca377e8","name":"issue-2510","id":"rco12lq2oq6j3HjAb1d7","createdBy":"00unkw1sfbTw08c0g1d7","created":"2025-10-12T21:10:39Z","lastUpdated":"2025-10-12T21:10:39Z","lastUpdatedBy":"00unkw1sfbTw08c0g1d7","_links":{"self":{"href":"https://oie-00-admin.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2oq6j3HjAb1d7","hints":{}}},"status":"INACTIVE","priority":0}'
+        headers:
+            Accept-Ch:
+                - Sec-CH-UA-Platform-Version
+            Content-Length:
+                - "620"
+            Content-Type:
+                - application/json
+            Date:
+                - Sun, 12 Oct 2025 21:10:39 GMT
+            Location:
+                - https://dcp-testing-oig-2025-06-26-admin.oktapreview.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2oq6j3HjAb1d7
+            Referrer-Policy:
+                - strict-origin-when-cross-origin
+        status: 201 Created
+        code: 201
+        duration: 2.214335041s
+    - id: 1
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
+        content_length: 0
+        host: oie-00.dne-okta.com
+        headers:
+            Accept:
+                - application/json
+            Authorization:
+                - SSWS REDACTED
+        url: https://oie-00.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2oq6j3HjAb1d7
+        method: GET
+      response:
+        proto: HTTP/2.0
+        proto_major: 2
+        proto_minor: 0
+        content_length: -1
+        uncompressed: true
+        body: '{"requesterSettings":{"type":"GROUPS","groups":[{"id":"00gqwtobh7LL5TuLp1d7"}]},"accessScopeSettings":{"type":"GROUPS","groups":[{"id":"00gouu5aq9Gq0JGLH1d7"}]},"approvalSequenceId":"68d224058c0cff364ca377e8","name":"issue-2510","id":"rco12lq2oq6j3HjAb1d7","createdBy":"00unkw1sfbTw08c0g1d7","created":"2025-10-12T21:10:39Z","lastUpdated":"2025-10-12T21:10:39Z","lastUpdatedBy":"00unkw1sfbTw08c0g1d7","_links":{"self":{"href":"https://oie-00-admin.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2oq6j3HjAb1d7","hints":{}}},"status":"INACTIVE","priority":0}'
+        headers:
+            Accept-Ch:
+                - Sec-CH-UA-Platform-Version
+            Content-Type:
+                - application/json
+            Date:
+                - Sun, 12 Oct 2025 21:10:41 GMT
+            Referrer-Policy:
+                - strict-origin-when-cross-origin
+        status: 200 OK
+        code: 200
+        duration: 1.31930275s
+    - id: 2
+      request:
+        proto: HTTP/1.1
+        proto_major: 1
+        proto_minor: 1
+        content_length: 0
+        host: oie-00.dne-okta.com
+        headers:
+            Accept:
+                - application/json
+            Authorization:
+                - SSWS REDACTED
+        url: https://oie-00.dne-okta.com/governance/api/v2/resources/0oaqgxmg2n2FjHLzw1d7/request-conditions/rco12lq2oq6j3HjAb1d7
+        method: DELETE
+      response:
+        proto: HTTP/2.0
+        proto_major: 2
+        proto_minor: 0
+        content_length: 0
+        body: ""
+        headers:
+            Accept-Ch:
+                - Sec-CH-UA-Platform-Version
+            Date:
+                - Sun, 12 Oct 2025 21:10:43 GMT
+            Referrer-Policy:
+                - strict-origin-when-cross-origin
+        status: 204 No Content
+        code: 204
+        duration: 1.314347541s

